pkg/reconciler/domainmapping/reconciler.go

```diff
@@ -322,6 +316,34 @@ func (r *Reconciler) resolveRef(ctx context.Context, dm *v1alpha1.DomainMapping)
 	return resolved.Host, parts[0], nil
 }
 
+func httpOption(ctx context.Context, annotations map[string]string) (netv1alpha1.HTTPOption, error) {
+	// Set HTTPOption via domainmapping annotations.
+	if len(annotations) != 0 && networking.GetHTTPProtocol(annotations) != "" {
+		protocol := strings.ToLower(networking.GetHTTPProtocol(annotations))
+		switch networkingpkg.HTTPProtocol(protocol) {
+		case networkingpkg.HTTPEnabled:
+			return netv1alpha1.HTTPOptionEnabled, nil
+		case networkingpkg.HTTPRedirected:
+			return netv1alpha1.HTTPOptionRedirected, nil
+		default:
+			return "", fmt.Errorf("incorrect http-protocol annotation: " + protocol)
+		}
+	}
+
+	// Set HTTPOption via config-network.
+	switch config.FromContext(ctx).Network.HTTPProtocol {
+	case networkingpkg.HTTPEnabled:
+		return netv1alpha1.HTTPOptionEnabled, nil
+	case networkingpkg.HTTPRedirected:
+		return netv1alpha1.HTTPOptionRedirected, nil
+	// This will be deprecated soon
+	case networkingpkg.HTTPDisabled:
+		return "", nil
+	default:
+		return "", nil
```

Comment:
should we return error here?

Reply:
I ref the smae impl here https://github.com/knative/serving/blob/knative-v1.3.0/pkg/reconciler/route/resources/ingress.go#L226

Reply:
I want to add return error after the HTTPDisabled option is completely deprecated

Reply:
Or do you have any ideas?

Reply:
Ideally, we should return error as Zhimin said. But let's keep the same implementation with Route's ingress until we can clean up `HTTPDisable` option.

Reply:
I understand, then I will not add the return error message for now.
thx @nak3 @ZhiminXiang

Reply:
I am fine with it.  could we add a log for those two scenarios then (HTTPDisabled and default)?

Reply:
Hi @ZhiminXiang 
I added logs

Reply:
plz check